File: IntegrationMod/src/main/java/com/example/examplemod/GetSkinRequest.java
package com.example.examplemod;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.util.stream.*;

public final class GetSkinRequest extends Message{

    private String _nickname;

    public GetSkinRequest(String nickname){
        this._nickname = nickname;
    }

    @Override
    public MessageHeader getMessageHeader() throws Exception {
        return new MessageHeader(MessageRegistry.GetKeyForMessageType(GetSkinRequest.class), (Integer.SIZE / 8) + _nickname.length());
    }

    @Override
    public void ApplyData(Stream stream) {

    }

    @Override
    public byte[] getData() throws Exception {
        MessageHeader header = getMessageHeader();
        ByteBuffer byteBuffer = ByteBuffer.allocate(header.DataLength);
        byteBuffer.order(ByteOrder.LITTLE_ENDIAN);
        byteBuffer.putInt(_nickname.length());
        byteBuffer.put(_nickname.getBytes());
        return byteBuffer.array();
    }
}

File: Common/Network/Messages/GetSkin/GetSkinRequest.cs
namespace Common.Network.Messages.GetSkin;

public sealed class GetSkinRequest: Message
{
    public string Nickname;

    public GetSkinRequest() { }
    
    public GetSkinRequest(string nickname)
    {
        Nickname = nickname;
    }

    public override MessageHeader GetHeader()
    {
        return new MessageHeader(MessageRegistry.GetKeyForMessageType(typeof(GetSkinRequest)), Nickname.Length);
    }

    public override void ApplyData(Stream stream)
    {
        Nickname = StringReadStream(stream);
    }

    protected override Stream GetData()
    {
        MemoryStream buffer = new MemoryStream(GetHeader().DataLength);
        WriteToStream(buffer, Nickname);
            
        buffer.Flush();
        buffer.Position = 0;
        return buffer;
    }
}